### PR TITLE
Add stress-ng

### PR DESCRIPTION
**Pull request by mtekman**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

This adds stress-ng, a stress testing utility to conda

~~According to [the source](https://github.com/ColinIanKing/stress-ng) it relies on `libbsd` and `libipsec-mb-dev` and `libsctp-dev` which do not appear to exist in conda.~~

~~However, the [Arch linux PKGBUILD](https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=stress-ng) does not appear to declare any such dependencies, so I'm crossing my fingers and hoping it works without them.~~

- ~~Currently failing at~~
 ```
RuntimeError: recipe ('build_artifacts', 'stress-ng_1613921675144', '_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p', 'info', 'recipe', 'meta.yaml') in wrong directory
```

Edit: currently builds, and `libbsd` is not necessary.

### CODE REVIEW

File: recipes/stress-ng/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
+{% set name = "stress-ng" %}
+{% set version = "0.12.02" %}
+
+package:
+  name: "{{ name|lower }}"
+  version: "{{ version }}"
+
+source:
+  url: https://kernel.ubuntu.com/~cking/tarballs/stress-ng/stress-ng-0.12.02.tar.xz
```
Comment (bgruening) [inline recipes/stress-ng/meta.yaml:9]: can you use a release from https://github.com/ColinIanKing/stress-ng/releases ?

Comment (bgruening) [reply recipes/stress-ng/meta.yaml:9]: You should use {{ version }} here as well.

File: recipes/stress-ng/build.sh [hunk 2]
```diff
@@ -0,0 +1,4 @@
+
+make && make install
```
Comment (bgruening) [inline recipes/stress-ng/build.sh:2]: make install into which directory? Are you sure you don't need to specify $PREFIX

Comment (bgruening) [reply recipes/stress-ng/build.sh:2]: You can run make lite-test or make test here as well

**Review by bgruening — COMMENTED** [review]
the error looks like a glib one. So maybe try an older version of this tool or depend on glib?

**Comment by bgruening** [comment]
Isn't the error:

```
stress-exec.c:318:26: error: 'O_PATH' undeclared (first use in this function); did you mean 'DT_RPATH'?
  318 |      fd = open(filename, O_PATH);
      |                          ^~~~~~
      |                          DT_RPATH
stress-exec.c:318:26: note: each undeclared identifier is reported only once for each function it appears in
make[1]: *** [Makefile:371: stress-exec.o] Error 1
make[1]: Leaving directory '$SRC_DIR'
make: *** [Makefile:355: all] Error 2

```

**Comment by mtekman** [comment]
@bgruening I'm out of ideas -- still getting that O_PATH error, but not sure how to resolve it. Apparently this is offered by the kernel, and that glibc can offer it, but I am at a loss as to how to tell `make` to use it. I would think `autogen` would detect glib automatically if declared in the build environment, but this does not seem to be the case.

**Comment by mtekman** [comment]
@bgruening it now builds thanks to the upstream author, and the `make test` tests work, but for some reason the binary cannot be found during the custom tests. Any idea here?

**Comment by mtekman** [comment]
The build check seems to be acting strangely -- errors I was getting before relating to missing libraries during runtime no longer seem to happen, and attempts to break the package meta.yaml still resulted in a working config.

For now I revert back to the config that made the most sense to me. I also wait for the upstream author to make an official minor release (he is very quick with replies) as the current meta is pinned to a commit than an official tarball.

**Review by bgruening — APPROVED** [review]

**Comment by mtekman** [comment]
Oh wow, nice edits -- I actually tried a `BINDIR=""` build which did not work, didn't think to replace it in the Makefile. Also pleasantly surprised to see most of the dependencies are skipped

**Pull request merged**